src/hotspot/share/prims/jvm.cpp

```diff
@@ -2124,7 +2124,7 @@ JVM_ENTRY(jobjectArray, JVM_GetPermittedSubclasses(JNIEnv* env, jclass current))
   Klass* c = java_lang_Class::as_Klass(mirror);
   assert(c->is_instance_klass(), "must be");
   InstanceKlass* ik = InstanceKlass::cast(c);
-  {
+  if (ik->is_sealed()) {
     JvmtiVMObjectAllocEventCollector oam;
     Array<u2>* subclasses = ik->permitted_subclasses();
     int length = subclasses == NULL ? 0 : subclasses->length();
```

Comment:
Minor comment - you don't really need the check of subclasses == NULL here since subclasses will never be NULL.  You could just assign length to subclasses->length();

Reply:
+1.   is_sealed returns true iff `_permitted_subclasses != NULL`

Reply:
Thanks for the reviews.  I removed the check of subclasses == NULL in the updated commit.